File: engine/b/src/main/java/net/crazyinvestor/engine_b/config/BithumbWSConnectionConfiguration.java
package net.crazyinvestor.engine_b.config;

import net.crazyinvestor.engine_b.enums.BithumbSymbol;
import net.crazyinvestor.engine_b.enums.BithumbTickTypes;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.socket.client.ReactorNettyWebSocketClient;
import org.springframework.web.reactive.socket.client.WebSocketClient;

import java.net.URI;
import java.util.LinkedList;
import java.util.List;

@Configuration
public class BithumbWSConnectionConfiguration {
    private final String baseURL;

    public BithumbWSConnectionConfiguration(
            @Value("${app.exchange.uri.bithumb}") final String baseURL
    ) {
        this.baseURL = baseURL;
    }

    @Bean
    @Qualifier("bithumb")
    public WebSocketClient getWebsocketClient() {
        return new ReactorNettyWebSocketClient();
    }

    @Bean
    @Qualifier("bithumb")
    public URI getBithumbBaseURI() {
        return URI.create(baseURL);
    }

    @Bean
    public List<BithumbSymbol> getBithumbWebsocketSubscribeSymbolList() {
        List<BithumbSymbol> symbols = new LinkedList<>();
        symbols.add(BithumbSymbol.BTC_KRW);
        symbols.add(BithumbSymbol.ETH_KRW);

        return symbols;
    }

    @Bean
    public List<BithumbTickTypes> getBithumbWebsocketSubscribeTickTypeList() {
        List<BithumbTickTypes> tickTypes = new LinkedList<>();
        tickTypes.add(BithumbTickTypes.HR_1);

        return tickTypes;
    }

}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/config/BithumbWSConnectionConfiguration.kt
package net.crazyinvestor.ticker_producer_engine.config

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.socket.client.ReactorNettyWebSocketClient
import org.springframework.web.reactive.socket.client.WebSocketClient
import java.net.URI
import java.util.*

import net.crazyinvestor.ticker_producer_engine.enums.BithumbSymbol
import net.crazyinvestor.ticker_producer_engine.enums.BithumbTickTypes

@Configuration
class BithumbWSConnectionConfiguration(
    @Value("\${app.exchange.uri.bithumb}") val baseURL: String,
) {

    @Bean
    @Qualifier("bithumbClient")
    fun websocketClient(): WebSocketClient {
        return ReactorNettyWebSocketClient()
    }

    @Bean
    @Qualifier("bithumb")
    fun bithumbBaseURI(): URI {
     return URI.create(baseURL)
    }

    @Bean
    fun bithumbWebsocketSubscribeSymbolList(): List<BithumbSymbol> {
        val symbols: MutableList<BithumbSymbol> = LinkedList<BithumbSymbol>()
        symbols.add(BithumbSymbol.BTC_KRW)
        symbols.add(BithumbSymbol.ETH_KRW)
        return symbols.toList()
    }

    @Bean
    fun bithumbWebsocketSubscribeTickTypeList(): List<BithumbTickTypes> {
        val tickTypes: MutableList<BithumbTickTypes> = LinkedList<BithumbTickTypes>()
        tickTypes.add(BithumbTickTypes.HR_1)

        return tickTypes.toList()
    }
}